numpy/core/tests/test_datetime.py

```diff
@@ -2279,6 +2279,95 @@ def test_limit_str_roundtrip(self, time_unit, sign):
         limit_via_str = np.datetime64(str(limit), time_unit)
         assert limit_via_str == limit
 
+    def test_datetime_hash(self):
+        assert hash(np.datetime64()) == -2  # NaT
+
+        dt = np.datetime64(2348, 'W')  # 2015-01-01
+        for unit in ('Y', 'M', 'W', 'D', 'h', 'm', 's', 'ms', 'us'):
+            dt2 = np.datetime64(dt, unit)
+            assert dt == dt2
+            assert hash(dt) == hash(dt2)
+            assert dt2 in {dt}
+
+            dt3 = np.datetime64(int(dt2.astype(int)) + 1, unit)
+            assert hash(dt) != hash(dt3)  # doesn't collide
+
+        for unit in ('h', 'm', 's', 'ms', 'us'):
+            dt2 = np.datetime64(dt, unit)
+            pydt = dt2.astype(datetime.datetime)
+            assert isinstance(pydt, datetime.datetime)
+            assert pydt == dt2
+            assert hash(pydt) == hash(dt2)
+            assert dt2 in {pydt}
+
+        dt = np.datetime64(-102894, 'W')  # -002-01-01
+        for unit in ('Y', 'M', 'W', 'D', 'h', 'm', 's', 'ms', 'us'):
+            dt2 = np.datetime64(dt, unit)
+            assert dt == dt2
+            assert hash(dt) == hash(dt2)
+            assert dt2 in {dt}
+
+        dt = np.datetime64(3, 'ms')
+        for unit in ('ms', 'us', 'ns', 'ps', 'fs', 'as'):
+            dt2 = np.datetime64(dt, unit)
+            assert hash(dt) == hash(dt2)
+            assert dt2 in {dt}
+
+            dt3 = np.datetime64(int(dt2.astype(int)) + 1, unit)
+            assert hash(dt) != hash(dt3)  # doesn't collide
+
+        for wk in range(500000, 500010):  # 11552-09-04
+            dt = np.datetime64(wk, 'W')
+            for unit in ('W', 'D', 'h', 'm', 's', 'ms', 'us'):
+                dt2 = np.datetime64(dt, unit)
+                assert hash(dt) == hash(dt2)
+
+    def test_timedelta_hash(self):
+        assert_raises(ValueError, hash, np.timedelta64(123))  # generic
+
+        td = np.timedelta64(45, 'Y')
+        for unit in ('Y', 'M'):
+            td2 = np.timedelta64(td, unit)
+            assert td == td2
+            assert hash(td) == hash(td2)
+            assert td2 in {td}
+
+        td = np.timedelta64(10, 'W')
+        for unit in ('W', 'D', 'h', 'm', 's', 'ms', 'us'):
+            td2 = np.timedelta64(td, unit)
+            assert td == td2
+            assert hash(td) == hash(td2)
+            assert td2 in {td}
+
+            td3 = np.timedelta64(int(td2.astype(int)) + 1, unit)
+            assert hash(td) != hash(td3)  # doesn't collide
+
+            pytd = td2.astype(datetime.timedelta)
+            assert isinstance(pytd, datetime.timedelta)
+            assert hash(pytd) == hash(td2)
+            # Pure Python implementation of timedelta.__eq__ should
+            # raise NotImplemented instead of returning False.
+            # (https://bugs.python.org/issue37579)
+            # This is not yet backported to PyPy.
+            if not IS_PYPY:
+                assert pytd == td2
+                assert td2 in {pytd}
```

Comment:
This should not be needed now. CI uses the latest build, which now includes a fix.

Reply:
@mattip. Doh! Will fix...

Reply:
Sorry for the mess.

Reply:
Haha no problemo! I'm glad I didn't have to investigate it myself. Thanks.